### PR TITLE
fix: Add shaded jar for java-client and document its use

### PR DESCRIPTION
**Pull request by jnh5y**
Addresses https://github.com/confluentinc/ksql/issues/7470 and https://github.com/confluentinc/ksql/issues/9245.

### Testing done 
I have tested locally with some example Java client code.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by JimGalasyn — APPROVED** [review]
LGTM!

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 1]
```diff
@@ -113,6 +114,12 @@ Start by creating a `pom.xml` for your Java application:
       by replacing the repositories in the example POM above with a repository with this
       URL instead. Also update `ksqldb.version` to be a CP version, such as `{{ site.cprelease }}`, instead.
 
+!!! note
+      The `with-dependencies` artifact was introduced in ksqlDB version 0.29 and CP version 7.4.0.  This jar includes  
+      all the necessary dependencies and relocates most of them in an attempt to avoid classpath issues.
+      If you want more control over the dependencies on the classpath, you can depend directly on the client
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:120]: How does this work exactly? Does this mean the user would have to ensure that all the Java client's dependencies are available on the classpath via other methods? When would a user prefer to do that rather than using the shaded dependencies, given that those dependencies are already shaded in order to avoid name conflicts?

**Review by vcrfxia — APPROVED** [review]
Thanks @jnh5y -- LGTM with a question inline!

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 2]
```diff
@@ -113,6 +114,12 @@ Start by creating a `pom.xml` for your Java application:
       by replacing the repositories in the example POM above with a repository with this
       URL instead. Also update `ksqldb.version` to be a CP version, such as `{{ site.cprelease }}`, instead.
 
+!!! note
+      The `with-dependencies` artifact was introduced in ksqlDB version 0.29 and CP version 7.4.0.  This jar includes
```
Comment (agavra) [inline docs/developer-guide/ksqldb-clients/java-client.md:118]: I'm having problems groking this:

> you can depend directly on the client by removing that classifier from the dependency block.

Are you saying that if you have vertx pinned in your own project, and you don't want to take it from `ksqldb-api-client` then you can _remove_ the `with-dependencies` classifier? That makes sense, but for some reason I'm struggling with the english.

**Review by agavra — APPROVED** [review]
Thanks for sleuthing through shading docs to get this out!

File: docs/developer-guide/ksqldb-clients/java-client.md (hunk 1 above)
Comment (jnh5y) [reply docs/developer-guide/ksqldb-clients/java-client.md:120]: If a user depends on `io.confluent.ksql:ksqldb-api-client` without the `with-dependencies` classifier (e.g.)
```        <dependency>
            <groupId>io.confluent.ksql</groupId>
            <artifactId>ksqldb-api-client</artifactId>
            <version>${ksqldb.version}</version>
        </dependency>```
then yes, they would be responsible for getting all the other dependencies.

A user may prefer to do that if their Java project was using one or more of the bundled dependencies and they wanted to use that dependency as common code.  They may wish to do that to reduce the size of the classpath; that could matter if one were building up a Docker image.

Basically, they'd be taking on the risk of classpath conflicts in order to reduce classpath size.

File: docs/developer-guide/ksqldb-clients/java-client.md (hunk 2 above)
Comment (jnh5y) [reply docs/developer-guide/ksqldb-clients/java-client.md:118]: Yeah, English is hard. 

I'm trying to suggest that folks could assemble their own classpath by depending on `io.confluent.ksql:ksqldb-api-client` directly.  

Victoria is basically asking the same question (I think).  Suggestions for how to clarify what I'm saying here?

Comment (agavra) [reply docs/developer-guide/ksqldb-clients/java-client.md:118]: I think an example would go a long way (e.g. if you want to do X do Y)

**Comment by yibinwang** [comment]
LGTM. @jnh5y can we merge this so that we can try it?

**Comment by jnh5y** [comment]
@agavra @vcrfxia I tried to write some new words.  Can you see if you like them?

**Review by vcrfxia — APPROVED** [review]
LGTM! Thanks @jnh5y .
